polling_stations/apps/data_finder/helpers/baked_data_helper.py

```diff
@@ -67,56 +50,45 @@ def __init__(self, elections_parquet_path: Path = None, **kwargs):
             )
         super().__init__(**kwargs)
 
-    def get_response_for_postcode(self, postcode: Postcode):
-        is_split, data_for_postcode = self.get_ballot_list(postcode)
-        data = {
-            "address_picker": is_split,
-            "addresses": [],
-        }
+    def get_response(self, postcode: Postcode, uprn: Optional[str] = None):
+        result = self.get_ballot_list(postcode, uprn)
 
-        if is_split:
-            data["addresses"] = data_for_postcode
-        else:
-            data["dates"] = self.ballot_list_to_dates(data_for_postcode)
+        if not result["address_picker"]:
+            result["ballots"] = self.get_full_ballots(result["ballot_ids"])
+            # TODO: how to set request_success if some requests failed here?
 
-        return data
+        return result
 
-    def get_full_ballots(self, ballot_data):
+    def get_full_ballots(self, ballot_ids):
         result = []
-        for ballot in ballot_data:
-            ballot_id = ballot["ballot_paper_id"]
+        for ballot_id in ballot_ids:
             url = ballot_paper_id_to_ee_url(ballot_id)
             req = session.get(url)
             result.append(req.json())
         return result
```

Comment:
There are several problems to note here:

1. There is no error handling at all here. Only happy path handling has been implemented.
2. These requests are being done in sequence, not in parallel. This means if the user has 4 upcoming ballots we make 4 HTTP requests one after another. We're re-using the same connection, but fundamentally it is still sequential.

This does need improving, but I'm going to deal with this topic as a follow up PR.